src/user-data.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file User data storage.
+ * @author Johan Nordberg <johan@steemit.com>
+ */
+
+import {JsonRpcError, JsonRpcMethodContext as JCtx} from '@steemit/jsonrpc'
+import {ValidationError} from 'sequelize'
+
+import {User} from './database'
+
+interface UserData {
+    email: string
+    phone: string
+}
+
+export async function getUserData(this: JCtx, account: string) {
```

Comment:
am i missing something here? why are you passing an argument called `this`?

Reply:
That's just a typescript-ism that let's you define a type for the context

Reply:
I'm not going to say the execution is cool, but it is a neat capability.

Reply:
Why is `this` being used at all in a function that isn't a member of an object?

Reply:
if your function isn't explicitly a member of a prototype, isn't it an antipattern to be using `this`? wouldn't you be better off passing your `JCtx` explicitly as the first argument to the function and naming it something like `ctx` or `context`?

Reply:
Because rpc method handlers have a context bound to them

Reply:
See https://github.com/steemit/koa-jsonrpc/blob/master/src/jsonrpc.ts

Reply:
> so that rpc methods can be added without having to specify a list of param names

can you articulate this a bit more? if `this` weren't bound, what would be different about the RPC methods?

Reply:
JSON-RPC 2.0 allows both positional and keyword arguments, javascript has only positional. To be able to call methods with both I need the list of param names. I'm parsing that from the functions toString() to provide a clean API where the function arguments are the RPC params and the request context is bound to the method. 

This isn't relevant to this PR though, if you wish to review my design decisions on how the jsonrpc service is implemented lets take that discussion somewhere else

Reply:
```
I'm parsing that from the functions toString()
```

Okay, we're going to have a long conversation about this. This type of introspection is catastrophically dangerous, especially when you're running through multiple layers of transpilation. Omitted until later I suppose.

Reply:
I understand your concern but it's very safe the way I'm doing it, let's put this up on the table then ;)